### PR TITLE
storageconsumer: Fix nil map and subvolume group naming

### PR DESCRIPTION
**Pull request by kesavanvt**
fix nil map and subvolume group naming

closes : #1521 
Signed-off-by: kesavan <kvellalo@redhat.com>

### CODE REVIEW

File: controllers/storageconsumer/storageconsumer_controller.go [hunk 1]
```diff
@@ -125,7 +126,7 @@ func (r *StorageConsumerReconciler) initReconciler(request reconcile.Request) {
 	r.cephBlockPool.Namespace = r.namespace
 
 	r.cephFilesystemSubVolumeGroup = &rookCephv1.CephFilesystemSubVolumeGroup{}
-	r.cephFilesystemSubVolumeGroup.Name = fmt.Sprintf("%s-%s", "cephFilesystemSubVolumeGroup", r.storageConsumer.Name)
+	r.cephFilesystemSubVolumeGroup.Name = fmt.Sprintf("%s-%s", "cephfilesystemsubvolumegroup", r.storageConsumer.Name)
```
Comment (Madhu-1) [inline controllers/storageconsumer/storageconsumer_controller.go:129]: you also need to set the filesystem name or else it will fail. @leseb am I correct?

Comment (leseb) [reply controllers/storageconsumer/storageconsumer_controller.go:129]: Correct, `FilesystemName` is a mandatory field.

Comment (kesavanvt) [reply controllers/storageconsumer/storageconsumer_controller.go:129]: This has been taken care in the `reconcileCephFilesystemSubVolumeGroup` method

**Comment by sabose** [comment]
@Madhu-1 @leseb @jarrpa could you review and merge please?

**Review by Madhu-1 — COMMENTED** [review]
LGTM

**Review by leseb — APPROVED** [review]

**Review by agarwal-mudit — APPROVED** [review]

**Comment by agarwal-mudit** [comment]
/cherry-pick release-4.10

@kesavanvt do you have a downstream BZ for this?

**Pull request merged**